packages/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerDetectorView.kt

```diff
@@ -149,12 +150,18 @@ class RNGestureHandlerDetectorView(context: Context) : ReactViewGroup(context) {
     }
   }
 
-  private fun tryAttachNativeHandlersToChildView(childId: Int) {
+  private fun tryAttachNativeHandlersToChildView(child: View) {
     val registry = RNGestureHandlerModule.registries[moduleId]
       ?: throw Exception("Tried to access a non-existent registry")
 
+    val id = if (child is ReactSwipeRefreshLayout) {
+      child.getChildAt(0).id
+    } else {
+      child.id
+    }
```

Comment:
1. Why do we need that? It looks like it makes attaching to `ReactSwipeRefreshLayout` impossible
2. Can this be done in the `NativeViewGestureHandlerHook`?

Reply:
>  It looks like it makes attaching to `ReactSwipeRefreshLayout` impossible

This should be handled by `attachVirtualChildren`, right?

> Can this be done in the `NativeViewGestureHandlerHook`?

We are inside NativeDetector, so  I don't exactly know what do you mean. Something like `handler.shouldAttachToChild` that would return `hook.shouldAttachToChild`?

Reply:
For now I can't see a clean way of doing it in hook. Added a comment in [7ad7ad7](https://github.com/software-mansion/react-native-gesture-handler/pull/3800/commits/7ad7ad70a9da17a29980bf2963edd259aaa7df55).